src/components/BlockDocumentsTable.vue

```diff
@@ -1,12 +1,23 @@
 <template>
   <div class="block-documents-table">
     <div class="block-documents-table__filters">
-      <ResultsCount label="Block" :count="total" class="block-documents-table__results" />
+      <span class="block-documents-table__results">
+        <ResultsCount label="Block" :count="total" />
+        <BlocksDeleteButton v-if="selectedBlockDocuments.length > 0" class="block-documents-table__delete" :selected="selectedBlockDocuments" small @delete="onDelete" />
+      </span>
       <SearchInput v-model="searchTerm" placeholder="Search blocks" label="Search blocks" class="block-documents-table__search" />
       <BlockSchemaCapabilitySelect v-model:selected="capabilities" class="block-documents-table__capability" />
       <BlockTypeSelect v-model:selected="blockTypes" class="block-documents-table__type" />
     </div>
     <p-table :data="blockDocuments" :columns="columns">
+      <template #selection-heading>
+        <p-checkbox v-model="model" @update:model-value="selectAllBlockDocuments" />
+      </template>
+
+      <template #selection="{ row }">
+        <p-checkbox v-if="row.can.delete" v-model="selectedBlockDocuments" :value="row.id" />
+      </template>
+
```

Comment:
`<p-table>` has built-in multi-select/checkboxes!

See https://github.com/PrefectHQ/prefect-ui-library/pull/2251

Reply:
Updated to use it!